google-cloud-datastore/src/main/java/com/google/cloud/datastore/testing/LocalDatastoreHelper.java

```diff
@@ -113,11 +115,95 @@ private LocalDatastoreHelper(double consistency, int port) {
     emulatorRunners = ImmutableList.of(gcloudRunner, downloadRunner);
   }
 
+  /** A builder for {@code LocalDatastoreHelper} objects. */
+  public static class Builder {
+    private List<EmulatorRunner> emulatorRunners;
+    private double consistency;
+    private int port;
+    private Path dataDir;
+    private boolean storeOnDisk = Boolean.TRUE.booleanValue();
+
+    private Builder() {}
+
+    public Builder setConsistency(double consistency) {
+      this.consistency = consistency;
+      return this;
+    }
+
+    public Builder setPort(int port) {
+      this.port = port;
+      return this;
+    }
+
+    public Builder setDataDir(Path dataDir) {
+      this.dataDir = dataDir;
+      return this;
+    }
+
+    public Builder setStoreOnDisk(boolean storeOnDisk) {
+      this.storeOnDisk = storeOnDisk;
+      return this;
+    }
+
+    /** Creates a {@code LocalDatastoreHelper} object. */
+    public LocalDatastoreHelper build() {
+      return new LocalDatastoreHelper(this);
+    }
+  }
+
+  LocalDatastoreHelper(Builder builder) {
+    super(
+        "datastore",
+        builder.port > 0 ? builder.port : BaseEmulatorHelper.findAvailablePort(DEFAULT_PORT),
+        PROJECT_ID_PREFIX + UUID.randomUUID().toString());
+    this.consistency = builder.consistency > 0 ? builder.consistency : DEFAULT_CONSISTENCY;
+    this.port = builder.port;
+    this.gcdPath = builder.dataDir;
+    this.storeOnDisk = builder.storeOnDisk;
+    String binName = BIN_NAME;
+    if (isWindows()) {
+      binName = BIN_NAME.replace("/", "\\");
+    }
+    List<String> gcloudCommand = new ArrayList<>(Arrays.asList(GCLOUD_CMD_TEXT.split(" ")));
+    gcloudCommand.add(GCLOUD_CMD_PORT_FLAG + "localhost:" + port);
+    gcloudCommand.add(CONSISTENCY_FLAG + builder.consistency);
+    if (!storeOnDisk) {
+      gcloudCommand.add("--no-store-on-disk");
+    }
+    GcloudEmulatorRunner gcloudRunner =
+        new GcloudEmulatorRunner(gcloudCommand, VERSION_PREFIX, MIN_VERSION);
+    List<String> binCommand = new ArrayList<>(Arrays.asList(binName, "start"));
+    binCommand.add("--testing");
+    binCommand.add(BIN_CMD_PORT_FLAG + port);
+    binCommand.add(CONSISTENCY_FLAG + consistency);
+    if (gcdPath != null) {
+      gcloudCommand.add("--data-dir=" + gcdPath.toString());
+    }
+    DownloadableEmulatorRunner downloadRunner =
+        new DownloadableEmulatorRunner(binCommand, EMULATOR_URL, MD5_CHECKSUM);
+    this.emulatorRunners = ImmutableList.of(gcloudRunner, downloadRunner);
+  }
+
+  /** Returns a builder for {@code LocalDatastoreHelper} object. */
+  public static LocalDatastoreHelper.Builder toBuilder() {
+    return new Builder();
+  }
```

Comment:
I guess toBuilder() should not be static and should return builder with current values of helper.